### PR TITLE
bug/TSK-809: Duplicated http call when selecting a task in workplaces

### PR DESCRIPTION
**Pull request by ignaciorecuerda**
- The call to get the classifications is just done once.
- Fix the problem when there are two classifications with the same name (now we compare with key property).
- Fix the problem that the list doesn't show every classification.
- Create a new component (dropdown)
- Change subscription to promise async/away

### CODE REVIEW

File: web/src/app/workplace/taskdetails/general/general-fields.component.ts [hunk 1]
```diff
@@ -47,20 +45,10 @@ export class TaskdetailsGeneralFieldsComponent implements OnInit, OnChanges {
     }
   }
 
-  ngOnInit() {
```
Comment (MiguelMartinRojas) [inline web/src/app/workplace/taskdetails/general/general-fields.component.ts]: Why did you move getClassifications from general-fields.component to task-details? I think that task-details doesn´t need to know about classifications.

File: web/src/app/workplace/taskdetails/taskdetails.component.ts [hunk 2]
```diff
@@ -63,6 +68,9 @@ export class TaskdetailsComponent implements OnInit, OnDestroy {
     this.masterAndDetailSubscription = this.masterAndDetailService.getShowDetail().subscribe(showDetail => {
       this.showDetail = showDetail;
     });
+    if (this.currentWorkbasket && !this.classifications) {
+      this.getClassificationByDomain();
```
Comment (MiguelMartinRojas) [inline web/src/app/workplace/taskdetails/taskdetails.component.ts]: This call is not needed

File: web/src/app/workplace/taskdetails/general/general-fields.component.ts (hunk 1 above)
Comment (MiguelMartinRojas) [reply web/src/app/workplace/taskdetails/general/general-fields.component.ts]: When you move back getClassification to this component, please make a test to check that classification service is called on init.

File: web/src/app/workplace/taskdetails/general/general-fields.component.ts [hunk 3]
```diff
@@ -21,22 +20,21 @@ export class TaskdetailsGeneralFieldsComponent implements OnInit, OnChanges {
   saveToggleTriggered: boolean;
   @Output() formValid: EventEmitter<boolean> = new EventEmitter<boolean>();
 
-  @Output() classificationsReceived: EventEmitter<Classification[]> = new EventEmitter<Classification[]>();
+  @Input()
+  classifications: Classification[];
```
Comment (MiguelMartinRojas) [inline web/src/app/workplace/taskdetails/general/general-fields.component.ts]: You don´t need it.

File: web/src/app/workplace/taskdetails/taskdetails.component.ts [hunk 4]
```diff
@@ -63,6 +68,9 @@ export class TaskdetailsComponent implements OnInit, OnDestroy {
     this.masterAndDetailSubscription = this.masterAndDetailService.getShowDetail().subscribe(showDetail => {
       this.showDetail = showDetail;
     });
+    if (this.currentWorkbasket && !this.classifications) {
```
[the remark this answers is not in the document]
Comment (ignaciorecuerda) [reply web/src/app/workplace/taskdetails/taskdetails.component.ts]: You are right

File: web/src/app/workplace/taskdetails/general/general-fields.component.ts (hunk 1 above)
Comment (ignaciorecuerda) [reply web/src/app/workplace/taskdetails/general/general-fields.component.ts]: done!

File: web/src/app/workplace/taskdetails/taskdetails.component.ts (hunk 2 above)
Comment (ignaciorecuerda) [reply web/src/app/workplace/taskdetails/taskdetails.component.ts]: done!

File: web/src/app/workplace/taskdetails/general/general-fields.component.ts (hunk 3 above)
Comment (ignaciorecuerda) [reply web/src/app/workplace/taskdetails/general/general-fields.component.ts]: done!
